Chap_API_Event.tex

```diff
@@ -105,12 +105,20 @@ \subsection{\code{PMIx_Register_event_handler}}
 
 If the \refarg{cbfunc} is non-\code{NULL}, the function call will be treated as a \emph{non-blocking} call and will return the following:
 
+\returnsimplenb
+The result of the registration operation shall be returned in the provided callback function along with the assigned event handler identifier.
+
+\returnstart
 \begin{itemize}
-\item \refconst{PMIX_SUCCESS} indicating that the request has been accepted for processing and the provided callback function will be executed upon completion of the operation. Note that the library must not invoke the callback function prior to returning from the \ac{API}. The result of the registration operation shall be returned in the provided callback function along with the assigned event handler identifier.
+% TODO: should there be a PMIX_OPERATION_SUCCEEDED here?
+\item \refconst{PMIX_OPERATION_SUCCEEDED}, returned only when \refarg{cbfunc} is non-\code{NULL} and the request was immediately processed successfully - the \refarg{cbfunc} will \textit{not} be called
+% TODO: What is an "undetermined reason"?  This either says that you should
+% never return an error other than this, or that this should only be returned if
+% your application has no idea what else to return.  Either way we should fix this.
```

Comment:
Not really - what it is saying is that the implementation doesn't provide a detail-specific error code for whatever went wrong. It could mean that there isn't a Standard error code that the implementer thought fully fit the reason for the problem, or some other reason.

Reply:
Guess I don't really care about retaining the "event registration" error. The generic "error" code will server just as well.

`PMIX_ERR_INIT` has a very specific meaning - namely, that you called an API prior to initializing the library. It therefore isn't associated with a given API as it can be returned from _any_ API.

Reply:
Ok.  The current description for PMIX_ERR_INIT is "Error during initialization."  I'm going to add a commit to change that to "The API requires that the PMIx library be initialized prior to being called."